src/helpers/contexts/GlobalNodeState.tsx

```diff
@@ -576,21 +575,25 @@ export const GlobalProvider = ({
   );
 
   const useInputData = useCallback(
-    (id: string, index: number) => {
+    // eslint-disable-next-line prefer-arrow-functions/prefer-arrow-functions, func-names
+    function <T extends InputValue>(
```

Comment:
Just curious why you did it like this

Reply:
From the perspective of `useInputData`. the input data is a `InputValue` (string or number (at least for now)), but from the perspective of e.g. `FileInput`, the input data has to be a string. Using generics, we resolve this perspective problem.

Reply:
Ah, about that. As it turns, TSX will interpret `<T>() => {}` as the HTML tag `<T>` followed by an arrow function. That's why I had to use `function`.

Reply:
Makes sense